test/app-tap/uri.test.lua

```diff
@@ -96,6 +96,76 @@ local function test_parse(test)
     test:is(tostring(error), expected_errmsg, "error message")
 end
 
+local function test_parse_ipv6_format_by_rfc2732(test)
+    -- The literal format of an IPv6 address can contain the following
+    -- characters [0-9;a-f;A-F]. It was discovered in that the
+    -- uri does not support the characters A-F, although it should
+    -- according to the RFC 2732.
+    test:plan(36)
+
+    local u
+
+    u = uri.parse("[FEDC:BA98:7654:3210:FEDC:BA98:7654:3210]")
+    test:is(u.host, 'FEDC:BA98:7654:3210:FEDC:BA98:7654:3210', 'host')
+    test:is(u.ipv6, 'FEDC:BA98:7654:3210:FEDC:BA98:7654:3210', 'ipv6')
```

Comment:
Use testcases-specific descriptions in assertions here and below.
For example: `host part in [FEDC:BA98:7654:3210:FEDC:BA98:7654:3210]` instead `host`.
Otherwise in TAP-report all testcases looks same.

Reply:
Replaced to luatest